### DISCUSSION TITLE
How to get selected item when row selected in DataGrid?

### DISCUSSION THREAD

**Discussion by LoveSpirals** [microsoft/fluentui#29123] [Q&A]
I see a `onSelectionChange` prop, but it returns a `DataGridProps` that ultimately returns a `TableRowId`.
The example shows mapping row Ids to items, but isn't there a way for the selection to tell me which item is selected?

**Comment by afreidz** [comment]
I worked around this by using the `getRowId` prop on `<DataGrid>` to set the `rowId` to some unique id in my data (rather than row index like it defaults to).  i then setup 3 states: 

1. the item data array used the render the table (may not need to be in state depending on your usecase)
1. the set of selected row ids like in the example
1. an array of selected items (items being of the same shape as the first state)

I then added a `useEffect` for when the selected row state changes, and filtered the first state by items with ids that matched the second state.  finally updated the third state with the filtered list of items.
